src/components/Appointments/AppointmentConfirmation.js

```diff
@@ -0,0 +1,194 @@
+import React, { useState, useEffect } from "react";
+import {
+  Container, Segment, Icon, Image, Button,
+} from "semantic-ui-react";
+import { useParams } from "react-router-dom";
+import styled from "styled-components";
+import { format, getHours, getMinutes } from "date-fns";
+
+import calendar from "images/calendar.png";
+import { getAppointmentBySlugPublic, updateStatusPublic } from "api/appointments";
+import Loader from "components/Loader";
+import { STATUS } from "utils/constants";
+
+const StyledContainer = styled(Container)`
+  && {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    margin: 3rem
+  }
+`;
+
+const StyledErrorContainer = styled(Container)`
+  && {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    margin: 3rem
+  }
+`;
+
+const StyledDiv = styled.div`
+  && {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    margin: 3rem;
+    border: 0.1rem solid lightgray;
+    border-radius: 0.3rem;
+    padding: 2rem;
+  }
+`;
+
+const StyledHeader = styled.label`
+  margin: 0.5rem;
+`;
+const StyledInfo = styled.label`
+  font-weight: bold;
+  font-size: 1.1rem;
+  margin-top: 0.5rem;
+`;
+
+const StyledDate = styled.label`
+  color: gray;
+  margin: 1.5rem;
+`;
+
+const ButtonDiv = styled.div`
+&& {
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
+  padding: 1rem;
+  gap: 1rem;
+}
+`;
+
+const AppointmentConfirmation = () => {
+  const [loading, setLoading] = useState(false);
+  const [isLoadingStatus, setIsLoadingStatus] = useState(null);
+  const [error, setError] = useState("");
+  const [status, setStatus] = useState(false);
+
+  const [appointment, setAppointment] = useState("");
+  const [patient, setPatient] = useState("");
+  const [doctor, setDoctor] = useState("");
+  const [time, setTime] = useState({});
+
+  const { slug } = useParams();
+
+  const formatTime = (hours, minutes) => {
+    const formattedHours = hours.toString().padStart(2, "0");
+    const formattedMinutes = minutes.toString().padStart(2, "0");
+    return `${formattedHours}:${formattedMinutes}`;
+  };
+
+  const handleAppointmentAction = async (s) => {
+    setIsLoadingStatus(s);
+    try {
+      const data = {
+        status: s,
+      };
+      await updateStatusPublic(slug, data);
+      setStatus(s);
+    } catch (er) {
+      setError("Unable to confirm appointment");
+    } finally {
+      setIsLoadingStatus(null);
+    }
+  };
+
+  useEffect(() => {
+    const fetch = async () => {
+      try {
+        setLoading(true);
+        const responseAppointment = await getAppointmentBySlugPublic(slug);
+        setAppointment(responseAppointment);
+        setPatient(responseAppointment.patient);
+        setDoctor(responseAppointment.doctor);
+        setStatus(responseAppointment.status);
```

Comment:
You can read `appointment.status`, no need for a separate state. If you need to update status, you can:
```
setAppointment({...appointment: status: s})
```